pkg/liveshare/port_forwarder_test.go

```diff
@@ -26,12 +26,26 @@ func TestNewPortForwarder(t *testing.T) {
 	}
 }
 
+type portUpdateNotification struct {
+	PortNotification
+	conn *jsonrpc2.Conn
+}
+
 func TestPortForwarderStart(t *testing.T) {
 	streamName, streamCondition := "stream-name", "stream-condition"
-	serverSharing := func(req *jsonrpc2.Request) (interface{}, error) {
+	const port = 8000
+	sendNotification := make(chan portUpdateNotification)
+	serverSharing := func(conn *jsonrpc2.Conn, req *jsonrpc2.Request) (interface{}, error) {
+		sendNotification <- portUpdateNotification{
+			PortNotification: PortNotification{
```

Comment:
The PortNotification field of the update struct is unused.

Reply:
We actually still need this. When we call ForwardToListener, ForwardToListener calls session.startSharing downstream, which waits for this notification. We're essentially simulating the notification that would usually be sent by a Codespace.